packages/filler/src/services/ContractInteractionService.ts

```diff
@@ -281,13 +285,100 @@ export class ContractInteractionService {
 			// Approve tokens if needed before estimating gas for failsafe
 			await this.approveTokensIfNeeded(order)
 
+			// For each output token, generate the storage slot of the token balance
+			// Balance slot depends upon implementation of the token contract
+			const overrides = (
+				await Promise.all(
+					order.outputs.map(async (output) => {
+						const tokenAddress = bytes32ToBytes20(output.token)
+						if (tokenAddress === ADDRESS_ZERO) return null
+
+						const userAddress = privateKeyToAddress(this.privateKey)
+						const testValue = toHex(maxUint256)
+
+						// Try both Solidity and Vyper implementations, both have different storage slot implementations
```

Comment:
Is there no standard Vyper or Solidity token contract used widely thatwe can derive the balance slot directly from without looping?

Reply:
There could be multiple ERC20 implementations in Solidity only. Basic Openzeppelin tokens without upgradeability have balances in slot 0. But with upgradeability the slot changes. 

And then there are other type of tokens which work as ERC20 token, they have different slot.

Thirdweb is also very popular token deployment protocol, their upgradeable erc20 token has balance in slot 151. 

I have optimised this loop with common slots. Which will most likely cover most cases. 

If not, then the loop will run. 

And also Vyper stores mappings in a different way. So it needs to be handled as well.

Reply:
Okay, sounds good.